Rewrite grid system and migrate static pages

jmalcic opened the pull request:
This pull resolves #250 and closes #499. It also decouples the CSS as in #485. The new grid system makes use of grid layout, which susbstantially reduces the number of divs necessary, and also makes extensive use of custom attributes rather than classes, which helps avoid naming conflicts and makes the HTML more readable. Finally, Stylus has been replaced with gulp-concat-css.

henrymercer commented on assets/styles/all-stylesheets.css:
```diff
@@ -0,0 +1,12 @@
+@import 'icons.css';
+@import 'mlh-trust-badge.css';
+
+/* @import 'schedule.styl'; */
```
Could you explain what this commented out code is for?

jmalcic replied:
The schedule.styl file is for the schedule page, which isn't part of the static site and so hasn't been migrated yet. I'd like to keep it in the manifest just to record it's there, then get rid of it when I migrate the schedule page.

henrymercer replied:
Please could you add in a comment explaining this?

jmalcic replied:
This should now be resolved

henrymercer commented on assets/styles/units.css:
```diff
@@ -0,0 +1,127 @@
+  --half-column: 0;
+}
+
+/* Hopefully, the following can eventually be replaced with `attr(rows)` and `attr(columns)` */
```
What's preventing us from doing this?

jmalcic replied:
Lack of implementation in any browsers

henrymercer commented on src/resources/faqs.yml:
```diff
@@ -1,70 +1,53 @@
 faqs:
   - question: What’s a hackathon?
     answer: A hackathon is a focused 24-hour event where you and a small team get to build something new and exciting, alongside a whole bunch of like-minded people.
-    rows: two half
```
The removal of these is great!

henrymercer commented on src/views/index.html:
```diff
@@ -4,194 +4,120 @@
-            </div>
+  <main layout="{{ layoutMode }}">
+    <div fixed grid="grid" sidebar columns="7.0" rows="7.0">
+      <img draggable="false" src="{{ asset('images/logo.svg') }}"></img>
```
I think the `img` closing tag here might be invalid HTML?

jmalcic replied:
Whoops good spot! Maybe I could add a linter to spot these?

henrymercer replied:
@jmalcic Think you fixed the other one in your new commit but this one's still incorrect

henrymercer replied:
Yeah, I wonder whether there's a way of linting the HTML when it has Nunjucks template tags within it.

jmalcic commented:
No major content changes, apart from where indicated in the commits. Specifically:

- The sponsorship page has been updated to be identical with this year's sponsorship document
- The content of the index page is mainly the same
  - I've replaced the link to the event page with a message saying it will be back soon
  - I've tweaked the message about subscribing for updates on applications
- I've changed the layout of the FAQs page to use text columns

henrymercer commented:
Adding as blocked since splash page has a layout issue in latest stable Safari (11.1.2)

jmalcic commented:
Safari 12 is out now for High Sierra, and Mojave is coming next Monday, @henrymercer can you verify the issue is fixed? And if so, shall we wait maybe a week to give people time to update Safari?

jmalcic commented:
@henrymercer I've altered the position of the first column (with the logo) so it now gets counted from the right, and then collapsed empty columns so everything gets aligned left. That should improve the display for very wide monitors. And sorry for all the pushes!

henrymercer commented:
I think it might be better to centre the main page on large screens, like we do for [the FAQs page](https://hackcambridge.com/faqs).  Also I think the logo is already quite large and noticeable, and I find it a bit distracting from the rest of the content that it sticks when you scroll the page.  Have also asked others on the team for feedback.
